src/VectorStore.ts

```diff
@@ -53,23 +53,40 @@ export class VectorStore {
 		tokensPerInterval: 20,
 		interval: "minute"
 	})
-	constructor(vault: Vault, createEmbeddingBatch: CreateEmbeddingFunction, createCompletion: CreateCompletionFunction) {
+
+	private exclusionPath: string
+
+	constructor(vault: Vault, createEmbeddingBatch: CreateEmbeddingFunction, createCompletion: CreateCompletionFunction, exclusionPath: string) {
 		this.vault = vault
 		this.createEmbeddingBatch = (textsToEmbed) => this.throttler.throttleCall(() => createEmbeddingBatch(textsToEmbed))
 		this.createCompletion = createCompletion
 		this.tokenizer = new GPT3Tokenizer({ type: "gpt3" })
+		this.exclusionPath = exclusionPath
+	}
+
+	setExclusionPath(exclusionPath: string) {
+		this.exclusionPath = exclusionPath
+	}
+
+	private isFilePathExcluded(file: string): boolean {
+		return (!!this.exclusionPath.trim() && file.startsWith(this.exclusionPath))
```

Comment:
do you know if its possible for this.exclusionPath to be null/undefined? to be safe, you may want to check that first
`this.exclusionPath && !!this.exclusionPath.trim() && file.startsWith(this.exclusionPath)`